cmd/crank/beta/validate/validate.go

```diff
@@ -0,0 +1,166 @@
+			if err != nil {
+				return nil, errors.Wrapf(err, "cannot derive composite CRD from XRD %q", xrd.GetName())
+			}
+			crds = append(crds, crd)
+
+			if xrd.Spec.ClaimNames != nil {
+				claimCrd, err := xcrd.ForCompositeResourceClaim(xrd)
+				if err != nil {
+					return nil, errors.Wrapf(err, "cannot derive claim CRD from XRD %q", xrd.GetName())
+				}
+				crds = append(crds, claimCrd)
+			}
+
+		case schema.GroupKind{Group: "pkg.crossplane.io", Kind: "Provider"}:
+			fmt.Println("Provider extension is not supported yet")
+			continue
+
+		case schema.GroupKind{Group: "pkg.crossplane.io", Kind: "Configuration"}:
+			fmt.Println("Configuration extension is not supported yet")
+			continue
+
+		default:
+			continue
+		}
+	}
+
+	return crds, nil
+}
+
+func newValidators(crds []*extv1.CustomResourceDefinition) (map[schema.GroupVersionKind]validation.SchemaValidator, error) {
+	validators := map[schema.GroupVersionKind]validation.SchemaValidator{}
+
+	for i := range crds {
+		internal := &ext.CustomResourceDefinition{}
+		if err := extv1.Convert_v1_CustomResourceDefinition_To_apiextensions_CustomResourceDefinition(crds[i], internal, nil); err != nil {
+			return nil, err
+		}
+
+		for _, ver := range internal.Spec.Versions {
+			var sv validation.SchemaValidator
+			var err error
+
+			if ver.Schema != nil && ver.Schema.OpenAPIV3Schema != nil {
+				sv, _, err = validation.NewSchemaValidator(ver.Schema.OpenAPIV3Schema)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			if internal.Spec.Validation != nil {
+				sv, _, err = validation.NewSchemaValidator(internal.Spec.Validation.OpenAPIV3Schema)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			validators[schema.GroupVersionKind{
+				Group:   internal.Spec.Group,
+				Version: ver.Name,
+				Kind:    internal.Spec.Names.Kind,
+			}] = sv
+		}
+	}
+
+	return validators, nil
+}
+
+func validateResources(resources []*unstructured.Unstructured, crds []*extv1.CustomResourceDefinition, skipSuccessLogs bool) error {
+	validators, err := newValidators(crds)
+	if err != nil {
+		return errors.Wrap(err, "cannot create validators")
+	}
+
+	failure, warning := 0, 0
+
+	for i, r := range resources {
+		v, ok := validators[r.GetObjectKind().GroupVersionKind()]
+		if !ok {
+			warning++
+			fmt.Println("[!] could not find CRD/XRD for: " + r.GroupVersionKind().String())
+			continue
+		}
+
+		re := v.Validate(&resources[i])
+		for _, e := range re.Errors {
+			failure++
+			fmt.Printf("[x] validation error %s - %s : %s\n", r.GroupVersionKind().String(), r.GetAnnotations()[composite.AnnotationKeyCompositionResourceName], e.Error())
+		}
+
+		if len(re.Errors) == 0 && !skipSuccessLogs {
+			fmt.Printf("[✓] %s - %s validated successfully\n", r.GroupVersionKind().String(), r.GetAnnotations()[composite.AnnotationKeyCompositionResourceName])
+		}
+	}
+
+	fmt.Printf("Validation complete with %d error, %d warning, %d success cases\n", failure, warning, len(resources)-failure-warning)
```

Comment:
we can consider removing the intro "Validation complete with" part of the string, so we immediately start with numbers:

```
1 error, 0 warning, 2 success cases
```